archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/ArchetypeEngineV2.java

```diff
@@ -91,6 +98,37 @@ public Path generate(InputResolver inputResolver,
         context.requireRootScope();
         onResolved.run();
 
+        //prepare user inputs for .helidon
+        Set<ContextValue.ValueKind> valueKindFilter = Set.of(ContextValue.ValueKind.EXTERNAL, ContextValue.ValueKind.USER);
+        Predicate<ContextEdge> contextFilter =
+                variation ->
+                        variation.value() != null
+                                && variation.node().path(true).startsWith("flavor")
+                                && valueKindFilter.contains(variation.value().kind());
+        Function<String, String> trimMapper = value -> {
+            Set<Character> forRemoval = Set.of('[', ']');
+            if (value == null || value.length() == 0) {
+                return value;
+            }
+            StringBuilder builder = new StringBuilder(value);
+            if (forRemoval.contains(builder.charAt(0))) {
+                builder.deleteCharAt(0);
+            }
+            if (builder.length() > 0 && forRemoval.contains(builder.charAt(builder.length() - 1))) {
+                builder.deleteCharAt(builder.length() - 1);
+            }
+            return builder.toString();
+        };
+        Map<String, String> values = ContextVisitor.process(context, contextFilter, ",");
+        context.scope().putValue(
```

Comment:
It's basically a map right ? Right now it is added as a context value as a list of entries as string separated by `=`.

The name `user-inputs` is special as it is provided by the engine,  maybe we should make it a little more special to emphasize that it is reserved. Also what happens if an input is declared with the id `user-inputs` ?

Instead of providing `user-inputs` as a built-in context value, we should provide it as a built-in model value.
I.e. `model.node().add()`

Reply:
Do we need to store these values in a `context` or in a `model` if decided to use separate option `--output-props-file` for the output file, so we can use the method properties.store(...) to write the content of this map to the file?

Reply:
Using model assumes that we create the file using the archetype. If we use an option like `--output-props-files` we would do the work in the engine or in the invoker.

That means either exposing the map for the invoker or the cli to create the file or passing `--output-props-file` to the engine for it to create the file.

`ArchetypeEngineV2.generate` is getting saturated so if we are going to add more arguments it may be a good opportinity to use a builder.

Reply:
Do you mean that `Builder` class should be implemented for  `ArchetypeEngineV2`? If it so `io.helidon.build.maven.archetype.postgenerate.ReflectedEngine` should be changed significantly. Is it suitable for us?

Reply:
It seems to be a necessary evil ; lets start doing it unless you see major issue(s).